app/subscriber/src/features/my-reports/ReportCard.tsx

```diff
@@ -73,7 +75,13 @@ export const ReportCard: React.FC<IReportCardProps> = ({ report, onDelete }) =>
           />
         </>
       }
-      onChange={(open) => open && fetchReport(report.id)}
+      onChange={(open) => {
+        const existingReport = myReports.find((r) => r.id === report.id);
+        // only need to fetch if we have fetched prior data but it is missing report instance data
+        if (open && !existingReport?.instances?.length) {
+          fetchReport(report.id);
+        }
+      }}
```

Comment:
Another optimization, whereas prior to this we were re-fetching the report data every time someone expanded the ReportCard, even if the report had already been fetched

Reply:
This should work for most cases, but doesn't account for the background processes which run auto-reports.